release.nix

```diff
@@ -178,15 +167,36 @@ let
         pkgsNew.pythonPackages.protobuf
         (oldAttrs : {doCheck = false; doInstallCheck = false;});
 
+    shell-hook-common =
+      pkgsNew.lib.optionalString pkgsNew.stdenv.isDarwin ''
+        export DYLD_LIBRARY_PATH=${pkgsNew.grpc}/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH
+      '' +
+      pkgsNew.lib.optionalString pkgsNew.stdenv.isLinux ''
+        export LD_LIBRARY_PATH=${pkgsNew.grpc}/lib''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
+      '';
```

Comment:
I know that you're just relocating this code, but "while you are here…"

I think you can use this instead:

https://github.com/NixOS/nixpkgs/blob/520da1d00f26d2f4d31eb3cd3c5dc8b3a1e998c6/pkgs/build-support/setup-hooks/fix-darwin-dylib-names.sh

In other words, if you add `pkgs.fixDarwinDylibNames` as a `nativeBuildInput` to the derivation then I think it does the right thing, but I haven't tested it, yet

Reply:
And make sure to guard `pkgs.fixDarwinDylibNames` behind `pkgs.stdenv.isDarwin`.